_gsocblogs/2022/blog_Phoenix_MohammadHumayunKhan.md

```diff
@@ -27,20 +27,38 @@ Most of my time hence was spent doing the aforementioned things. I read about Ty
+
+### Making sure that the new setup is stable
+
+> 27.08.2022 - 20.09.2022
+
+At this point, all the unit tests and e2e tests were running fine locally. But, I encountered a strange problem that the unit tests were failing on GitHub CI and not locally. First, I tried changing the environment in which I ran tests as our CI environment was on Ubuntu, then I tried installing the necessary packages as I used the [headless-gl](https://github.com/stackgl/headless-gl) library to mock WebGLRenderer but then, we decided to mock the WebGLRenderer completely as we intended to test only our code and not WebGLRenderer itself. 
+
+After this change, all the tests passed on the CI but still, I got a non-zero exit code on the execution of tests which also lead to a lot of confusion as we were not sure as to why we were getting this but [after a lot of digging](https://github.com/HSF/phoenix/pull/492/#issuecomment-1249671624), I got to know that it is due to the fact that Jest uses a single worker process while executing the tests on the CI and we just needed to add one more worker for it to stop showing the non-zero exit code. And, the tests then ran fine on the CI! ^_^
+
+This culminated in the end of the GSoC project timeline for me but I would be glad to continue working and add enhancements to Phoenix in the future as well. Thanks a lot to Edward, Fawad and HSF GSoC admins for selecting my project and giving me an opportunity to work here, at CERN-HSF. I am truly grateful. I would especially like to mention that the code reviews done by Fawad throughout the project work were actually fantastic and detailed. It led to a lot of improvements and I learned so many new things which would not have been the case otherwise so, kudos to that as well. 
 
-Currently, I am doing the migration of the unit tests inside [phoenix-ng](https://github.com/HSF/phoenix/blob/master/packages/phoenix-ng/README.md) package from Jasmine and Karma setup to use Jest as well. This will eventually help us to use a single test runner for all our unit tests and it will also be easily maintainable by the developers. I hope that using Jest, we can find solutions to a lot of problems that we faced earlier with unit tests in Phoenix.
+#### Relevant Links
 
-So, this is a short report of all the work that has been done in this project till the midterm evaluation. If you want to read in detail about the same, you can visit [my repo on GitHub](https://github.com/DamianArado/GSoC-2022-Phoenix/blob/main/PROGRESS.md).
+- Project Work Summary: https://github.com/DamianArado/GSoC-2022-Phoenix/blob/main/SUMMARY.md
+- Project Roadmap: https://github.com/DamianArado/GSoC-2022-Phoenix/blob/main/ROADMAP.md
+- Project Progress: https://github.com/DamianArado/GSoC-2022-Phoenix/blob/main/PROGRESS.md

```

Comment:
Please use `[Text](link)` format to make these links clickable

Reply:
Done. ✅